src/servant/tt_000/10E8.c

```diff
@@ -490,7 +491,128 @@ void func_80171ED4(s32 arg0) {
 
 INCLUDE_ASM("asm/us/servant/tt_000/nonmatchings/10E8", func_80172120);
 
-INCLUDE_ASM("asm/us/servant/tt_000/nonmatchings/10E8", func_80172C30);
+void func_80172C30(Entity* self) {
+    if (self->step == 1 && self->flags & FLAG_UNK_00200000) {
+        D_80174B38 = (self->ext.fam.cameraX - g_Tilemap.cameraX.i.hi) +
+                     (self->ext.fam.unkB0 - PLAYER.posX.i.hi);
+        D_80174B3C = (self->ext.fam.cameraY - g_Tilemap.cameraY.i.hi) +
+                     (self->ext.fam.unkB2 - PLAYER.posY.i.hi);
+
+        for (D_80174B30 = 0; D_80174B30 < 0x10; D_80174B30++) {
+            D_80174C3C[self->ext.fam.unk82][D_80174B30].x -= D_80174B38;
+            D_80174C3C[self->ext.fam.unk82][D_80174B30].y -= D_80174B3C;
+        }
+        return;
+    }
+
+    g_api.func_8011A3AC(self, 0, 0, &D_80174C30);
+    if (D_80174D3C != 0) {
+        self->zPriority = PLAYER.zPriority - 2;
+    }
+    switch (self->step) {
+    case 0:
+        func_801719E0(self);
+        if (self->ext.fam.unk82 == 0) {
+            func_8017160C(*(&D_80170660 + ((D_80174C30.unk0 / 10) * 5)), 0xD2);
```

Comment:
I think this is `func_8017160C(D_80170660[(D_80174C30.unk0 / 10) * 5], 0xD2);` https://decomp.me/scratch/zxlKC 
The *5 may indicate a struct, also this overlaps with some other data.

Reply:
Thanks! I was able to take the existing material and rearrange it a bit. We now have it as a 2D array of s32's. It might still be a struct, but for now the way we're indexing it seems to work. I brought this function up to match the others that touch that array, and moved the start of the array forward now that we see it should start at 0660.

Reply:
So actually, the resulting array length is not a multiple of 5. I think it actually starts earlier than what you have. If you look at this: https://decomp.me/scratch/1n0Qq

```
                    if (*(&D_80170658 + ((D_80174C30.unk0 / 10) * 0x14)) < (s16) temp_a0_2) {

...
        temp_v0_11 = func_80173F74((s16) temp_v0_10, arg0->ext.unkAC3C.unk86, *(&D_8017065C + ((D_80174C30.unk0 / 10) * 0x14)));
```

We don't have to get this 100% right now, just pointing it out.

Reply:
Yeah we will probably need to keep shifting stuff around. I think generally the data should match what has been decompiled - so once the function you linked is decompiled, the array can be adjusted in order to capture the full extent of the data. In general it seems that TT_000 is a bit messy, I think cleaning it up should be its own effort at some point. I don't want to have a PR for one function end up shifting too much to capture too much other stuff beyond the function's scope. Looks like there are only 3 functions left in TT_000, so might make sense for someone to finish those off and then make that whole overlay look as nice as possible.

Reply:
Actually... I'm already changing stuff, it looks clear that the start of the array should be at 80170658, let me try shifting it to capture all the data and if it's quick and easy I'll add it to this PR.

Reply:
I can't seem to get it to work. I make the table start at D_80170658, but the assembly for func_80172120 wants to reference D_8017065C (which is part of D_80170658). Then I get errors compiling with `undefined reference to D_8017065C` so I'm not sure how to address that. I think let's ignore that table for now and treat it as something to be resolved in parallel with func_80172120.